homeassistant/helpers/entity_platform.py

```diff
@@ -252,12 +258,15 @@ def _async_add_entity(self, entity, update_before_add, component_entities,
             raise HomeAssistantError(
                 'Invalid entity id: {}'.format(entity.entity_id))
         elif entity.entity_id in component_entities:
-            msg = 'Entity id already exists: {}'.format(entity.entity_id)
-            if entity.unique_id is not None:
-                msg += '. Platform {} does not generate unique IDs'.format(
-                    self.platform_name)
-            raise HomeAssistantError(
-                msg)
+            if replace:
```

Comment:
We should only ever replace when we find a device with the same unique id. We should never match on entity id.